File: src/main/java/hassan/personnel/managment/models/vm/PersonVm.java
package hassan.personnel.managment.models.vm;

import java.util.List;

/**
 * Created by Hassan on 11/21/2016.
 */
public class PersonVm {
    private int id;

    private String firstname;

    private String lastname;

    private PositionVm position;

    private List<WorkVm> works;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getFirstname() {
        return firstname;
    }

    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    public String getLastname() {
        return lastname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    public PositionVm getPosition() {
        return position;
    }

    public void setPosition(PositionVm position) {
        this.position = position;
    }

    public List<WorkVm> getWorks() {
        return works;
    }

    public void setWorks(List<WorkVm> works) {
        this.works = works;
    }
}

File: src/main/resources/static/app/models/entities/personVm.ts
///<reference path="workVm.ts"/>
///<reference path="positionVm.ts"/>
/**
 * Created by Hassan on 11/18/2016.
 */
class PersonVm {
    public id: number;
    public firstname: string;
    public lastname: string;
    public position: PositionVm;
    public works: Array<WorkVm>;


    constructor(id: number, firstname: string, lastname: string, position: PositionVm, works: Array<WorkVm>) {
        this.setup();
        if (id != undefined)this.id = id;
        if (firstname != undefined)this.firstname = firstname;
        if (lastname != undefined)this.lastname = lastname;
        if (position != undefined)this.position = position;
        if (works != undefined)this.works = works;
    }

    /** this function won't bind in angular */
    public getFullName() {
        return this.firstname
        + (Util.Utility.isNullOrUndefinedOrEmpty(this.lastname)
            ? ""
            : " " + this.lastname);
    }

    public get fullName():string{
        return this.getFullName();
    }

    private setup() {
        this.id = 0;
        this.firstname = null;
        this.lastname = null;
        this.position = null;
        this.works = [];
    }
}